build_dataset.py

```diff
@@ -141,26 +207,136 @@ def build_dataset():
         candidate_actions = act_gen.generate_actions(interactive_objs)
         diff2acts = find_valid_actions(env, state, candidate_actions)
 
+        valid_actions = diff2acts.values()
+
+        for v in valid_actions:
```

Comment:
Could we include this as an alternative function? I'm focused on trying to make sure things work just along the walkthrough trajectories for now.